### PR TITLE
Update labeler action wildcards

### PR DESCRIPTION
**Pull request by CaseyHillers**
https://github.com/flutter/flutter/issues/126002

## Pre-launch Checklist

- [X] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [X] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [X] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [X] I signed the [CLA].
- [X] I listed at least one issue that this PR fixes in the description above.
- [X] I updated/added relevant documentation (doc comments with `///`).
- [X] I added new tests to check the change I am making, or this PR is [test-exempt].
- [X] All existing and new tests are passing.

### CODE REVIEW

File: .github/labeler.yml [hunk 1]
```diff
@@ -72,35 +72,36 @@ engine:
...
   - any:
     - '**/pubspec.yaml'
     - '**/fix_data.yaml'
     - '**/*.expect'
     - '**/*test_fixes*'
-    - dev/**
-    - examples/**
-    - packages/flutter_goldens/**
-    - packages/flutter_goldens_client/**
+    - .github/**/*
```
Comment (CaseyHillers) [inline .github/labeler.yml:96]: TIL: These changes don't take affect in presubmit, which I found interesting.

**Review by keyonghan — COMMENTED** [review]
Not specific to this PR, does the `any` entries work? I spot this PR:
https://github.com/flutter/flutter/pull/127479

where the action bot removed `a:text input` and `framework` labels.

**Comment by CaseyHillers** [comment]
> Not specific to this PR, does the any entries work? I spot this PR:

Thanks for pointing this out! If I had to take a guess, these aren't truly regex. I've gone ahead and wrapped all the basic text searches with wildcards.

**Review by keyonghan — APPROVED** [review]
LGTM

**Pull request merged**